services/src/main/java/org/openlearn/service/FileInformationService.java

```diff
@@ -0,0 +1,129 @@
+
+	private final FileRepository fileRepository;
+
+	private final AssignmentRepository assignmentRepository;
+
+	private final PortfolioItemRepository portfolioItemRepository;
+
+	private final UserService userService;
+
+	private final FileInformationTransformer fileInformationTransformer;
+
+	private final StorageService storageService;
+
+	public FileInformationService(final FileRepository fileRepository,
+								  final AssignmentRepository assignmentRepository,
+								  final PortfolioItemRepository portfolioItemRepository,
+								  final UserService userService,
+								  final FileInformationTransformer fileInformationTransformer,
+								  final StorageService storageService) {
+		this.fileRepository = fileRepository;
+		this.portfolioItemRepository = portfolioItemRepository;
+		this.assignmentRepository = assignmentRepository;
+		this.userService = userService;
+		this.fileInformationTransformer = fileInformationTransformer;
+		this.storageService = storageService;
+	}
+
+	/**
+	 * Get all file information for a given assignment.
+	 *
+	 * @param assignmentId the id of the assignment
+	 * @param pageable the pagination information
+	 * @return the list of entities
+	 */
+	@Transactional(readOnly = true)
+	public Page<FileInformationDTO> findAllForAssignment(final Long assignmentId, Pageable pageable) {
+		log.debug("Request to get File Information for Assignment : {}", assignmentId);
+		Assignment assignment = assignmentRepository.findOne(assignmentId);
+		if (SecurityUtils.isCurrentUserInRole(AuthoritiesConstants.STUDENT)) {
+			return fileRepository.findByAssignmentAndUploadedByUser(assignment, userService.getCurrentUser(), pageable)
+				.map(fileInformationTransformer::transform);
+		} else {
+			return fileRepository.findByAssignment(assignment, pageable)
+				.map(fileInformationTransformer::transform);
+		}
+	}
+
+	/**
+	 * Get all file information for a given portfolio item.
+	 *
+	 * @param portfolioItemId the id of the portfolio item
+	 * @param pageable the pagination information
+	 * @return the list of entities
+	 */
+	@Transactional(readOnly = true)
+	public Page<FileInformationDTO> findAllForPorfolioItem(final Long portfolioItemId, Pageable pageable) {
+		log.debug("Request to get File Information for PortfolioItem : {}", portfolioItemId);
+		PortfolioItem portfolioItem = portfolioItemRepository.findOne(portfolioItemId);
+		return fileRepository.findByPortfolioItem(portfolioItem, pageable).map(fileInformationTransformer::transform);
+	}
+
+	/**
+	 * Get one file by id.
+	 *
+	 * @param id the id of the entity
+	 * @return the entity
+	 */
+	@Transactional(readOnly = true)
+	public FileInformationDTO findOne(final Long id) {
+		log.debug("Request to get File Information : {}", id);
+		return fileInformationTransformer.transform(fileRepository.findOne(id));
+	}
+
+	@Transactional(readOnly = true)
+	public String getFileNameFor(final Long id) {
```

Comment:
Didn't want to return `null` to avoid null pointer exception, and I wrote this before I had made the exception class. It should raise the exception though.